civis/ml/_helper.py

```diff
@@ -58,3 +62,261 @@ def list_models(job_type="train", author=SENTINEL, client=None, **kwargs):
                                         author=author,
                                         **kwargs)
     return models
+
+
+def put_models_shares_users(id, user_ids, permission_level,
+                            client=None,
+                            share_email_body='DEFAULT',
+                            send_shared_email='DEFAULT'):
+    """Set the permissions users have on this object
+
+    Use this on both training and scoring jobs.
+    If used on a training job, note that "read" permission is
+    sufficient to score the model.
+
+    Parameters
+    ----------
+    id : integer
+        The ID of the resource that is shared.
+    user_ids : list
+        An array of one or more user IDs.
+    permission_level : string
+        Options are: "read", "write", or "manage".
+    client : :class:`civis.APIClient`, optional
+        If not provided, an :class:`civis.APIClient` object will be
+        created from the :envvar:`CIVIS_API_KEY`.
+    share_email_body : string, optional
+        Custom body text for e-mail sent on a share.
+    send_shared_email : boolean, optional
+        Send email to the recipients of a share.
+
+    Returns
+    -------
+    readers : dict::
```

Comment:
I'm confused by the type annotation here. It looks like "users" and "groups" are lists of dicts, right? Should this say `list[dict]` to specify that explicitly?

Reply:
The `readers` is a single dictionary. It contains keys `users` and `groups`, each of which is a list of dictionaries. In Python notation, I think you're right that the types of `users` and `groups` would be `list[dict]`, but I think that what's show here is also correct, and consistent with the type notation used in the other API endpoints.